torch/csrc/api/src/nn/modules/adaptive.cpp

```diff
@@ -61,14 +61,40 @@ void AdaptiveLogSoftmaxWithLossImpl::reset_parameters() {
 }
 
 ASMoutput AdaptiveLogSoftmaxWithLossImpl::forward(const Tensor& input, const Tensor& target) {
-  TORCH_CHECK(input.size(0) == target.size(0),
-      "Input and target should have the same size in the batch dimension.");
+  auto targ_dim = target.dim();
+
+  TORCH_CHECK(
+    targ_dim == 1 || targ_dim == 0,
+    "0D or 1D target tensor expected, multi-target not supported");
+
+  if (targ_dim == 1) {
+    TORCH_CHECK(
+      input.dim() == 2,
+      "1D target tensor expects 2D input tensors, but found inputs with sizes ",
+      input.sizes(),
+      ".");
+  } else {
+    TORCH_CHECK(
+      input.dim() == 1,
+      "0D target tensor expects 1D input tensors, but found inputs with sizes ",
+      input.sizes(),
+      ".");
+  }
+
+  Tensor new_input, new_target;
+  if (targ_dim == 0) {
+    Tensor new_input = input.unsqueeze(0);
+    Tensor new_target = target.unsqueeze(0);
+  } else {
+    Tensor new_input = input;
+    Tensor new_target = target;
+  }
```

Comment:
Basically just created a new array if the input/target are 1D and 0D, respectively

Mainly did this because target_mask.nonzero() will return a (1, 0) tensor if target is a scalar, but this creates extra space.. not sure if this performance hit is acceptable

Reply:
I think the unsqueeze / re-squeeze approach is fine here. Note that unsqueeze just creates a view of the same underlying storage, so this shouldn't be too much of a perf hit